### ISSUE TITLE
[BUG] Issues with Sphinx 3.0+

### ISSUE THREAD

**Issue by cl-nkf** [jnikula/hawkmoth#17]
Firstly, thanks for your work on this project and I hope you're still continuing it.

My problem is the following. When doing this:
```C
/** test */
char test[10];
```

It outputs the following error:
```
WARNING: Error in declarator or parameters
Invalid C declaration: Expected identifier in nested name. [error at 5]
  char [10] test
  -----^
```

And generates this:
![image](https://user-images.githubusercontent.com/65283817/82231856-14849500-992e-11ea-9e90-e6aa631ad4ea.png)

Thank you for your answer.

**Comment by jnikula** [comment]
Thanks, I am still working on Hawkmoth, albeit slowly as of late.

I'm afraid I can't reproduce your issue. If I put this in `test.c`:
```C
/** test */
char test[10];
```
and run `python3 -m hawkmoth test.c` on it (this is for testing and bypasses Sphinx) I get:
```

.. c:var:: char [10] test

   test

```
without errors. What does it do for you?

**Comment by cl-nkf** [comment]
I've got the same results as yours.

**Comment by jnikula** [comment]
Are you using Sphinx version 3? Looks like there have been some backward incompatible changes.

**Comment by cl-nkf** [comment]
Ah yes, I didn't notice since I'm using the docker image.

**Comment by cl-nkf** [comment]
I went back to sphinx 2 and it's working. Thanks! 
By the way, if you're interested in updating the requirements, so far I'm using python3.8 and clang7 without any problems.

**Comment by Rahix** [comment]
@jnikula, I've also hit the warnings with sphinx 3.  I think the change is that sphinx now has separate directives for `c:struct`, `c:enum`, `c:unium` (See [The C Domain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-c-domain)).

**Comment by BrunoMSantos** [comment]
So the issue here is, I believe, with Sphinx taking issue with the placement of the `[10]` as part of the type rather than part of the variable name. I'm not sure, but I believe the expected way to do it with Sphinx 3 is:
```
.. c:var:: char test[10]
```
I've sent a patch for this; unsure if it breaks stuff in Sphinx 2, but it gets rid of the issue above. I also state there that it closes this ticket, but given the title change, I'm not sure that is 100% accurate.

Also sent a quick RFC for using the new directives. That one does break Sphinx 2 support for sure, at least in the current state, just looking for a reaction :)

**Comment by jnikula** [comment]
Should be fixed by pull #26 which includes @BrunoMSantos's fix as well as other sphinx 3 goodies.
